QualificationChecker/Controllers/QualificationClass.cs

```diff
@@ -0,0 +1,131 @@
+﻿using QualificationChecker.Models;
+
+namespace QualificationChecker.Controllers
+{
+    public abstract class QualificationManager
+    {
+        public Candidate Candidate { get; private set; }
+        public Organization Organization { get; private set; }
+        protected List<Candidate> QualifiedCandidates = new List<Candidate>();
+
+        protected QualificationManager(Candidate candidate, Organization organization)
+        {
+            Candidate = candidate;
+            Organization = organization;
+        }
+
+        // Method to tell if a Candidate is interested in the position
+        public bool IsCandidateInterestedInPosition()
+        {
+            if (Candidate.InterestedPositionIds == null)
+            {
+                return false;
+            }
+
+            return Candidate.InterestedPositionIds.Any(candidatePositionId =>
+                Organization.OrgQuestions.Any(OrgQuestion => OrgQuestion.PositionId == candidatePositionId));
+        }
+
+        // Method to make sure they meet the age requirement
+        public bool MeetsAgeRequirement()
+        {
+            return Candidate.Age >= Organization.MinAgeRequirement;
+        }
+
+        // Method to check if the answers match
+        private bool MatchAnswers(bool candidateAnswer, bool organizationAnswer)
+        {
+            // Compare each answer
+            return candidateAnswer == organizationAnswer;
+        }
+
+        public bool OrganizationFit()
+        {
+            return Candidate.OrganizationId == Organization.OrgId;
+        }
+
+        public virtual bool IsQualified()
+        {
+            return IsCandidateInterestedInPosition() && MeetsAgeRequirement() && OrganizationFit();
+        }
+
+        // Method to find qualified candidates
+        public void FindQualifiedCandidates(Candidate candidate)
+        {
+            {
+                if (CandidateMatches(candidate) && IsQualified())
+                {
+                    QualifiedCandidates.Add(candidate);
+                }
+            }
+
+        }
+
+        public bool CandidateMatches(Candidate candidate)
+        {
+            bool allAnswersMatch = true;
+
+            foreach (var candidateAnswer in candidate.CandidateQuestions)
```

Comment:
We are still trusting the candidate answers as the source of truth, instead of the organization.  I suspect the trouble you are having is having the `positionId` buried within the `candidateAnswer` ... maybe move that elsewhere?